new-packages/ts-project/src/index.ts

```diff
@@ -197,7 +189,24 @@ export function createProject(
         return extractMachineConfig(ctx, ts, call);
       });
     },
+    // TODO: consider exposing an object representing a file or a machine to the caller of the `extractMachines` and add a similar-ish method there
+    // for now we are just doing through the full extraction process again which is wasteful
```

Comment:
It's not only wasteful but also incorrect because we need to continue using the same `uniqueId`s and all. I'm working on this